### PR TITLE
rgw: init script in stop action waits until the radosgw stops

### PR DESCRIPTION
**Pull request by dmitryya**
I would like to propose the fix for  http://tracker.ceph.com/issues/11140

### CODE REVIEW

**Comment by liewegas** [comment]
looks okay to me!  this potentially means a very slow shutdown for rgw, but that's what force-stop is for.

File: src/init-radosgw.sysv [hunk 1]
```diff
@@ -109,6 +109,9 @@ case "$1" in
         #start-stop-daemon --stop -x $RADOSGW --oknodo
         killproc $RADOSGW
         echo "Stopping radosgw instance(s)..."
+        while pidof $RADOSGW >/dev/null; do
```
Comment (dachary) [inline src/init-radosgw.sysv]: the logic above should also be here, right ?

**Comment by dachary** [comment]
The logic looks good: for each radosgw client (as read from the ceph.conf list of sections), get the largest timeout and use that to wait when stopping the daemon. If the daemon does not stop within the required delay, just leave it. The alternative would be to kill it with -9 but that's what a forced stop would be about, as @liewegas mentionned.

**Comment by dmitryya** [comment]
Do I have to fix something or all fine?

**Comment by dachary** [comment]
There still is a pending question at https://github.com/ceph/ceph/pull/4056#discussion_r26689087

File: src/init-radosgw.sysv (hunk 1 above)
Comment (dmitryya) [inline src/init-radosgw.sysv]: you are right. update script with timeout logic.

**Comment by dmitryya** [comment]
May be add 'force-stop' action too. What do you think?

**Comment by dachary** [comment]
@dmitryya the force-stop action would make sense but it can be part of another pull request, don't you think ?

**Comment by dmitryya** [comment]
right.

**Comment by dachary** [comment]
@liewegas forgot your Reviewed-by:, sorry about that :-(

**Comment by liewegas** [comment]
no worries. thanks!

**Pull request merged**